src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -497,6 +499,22 @@ public static int Count<T>(this ImmutableArray<T> items, Func<T, bool> predicate
             return count;
         }
 
+        private static class UnderlyingArrayHelper<T>
+        {
+            public static readonly Func<object, object> GetValue =
+                typeof(ImmutableArray<T>).GetField("array",
+                    BindingFlags.Instance | BindingFlags.NonPublic).GetValue;
+        }
+
+        internal static T[] DangerousGetUnderlyingArray<T>(this ImmutableArray<T> array)
+            => (T[])UnderlyingArrayHelper<T>.GetValue(array);
+
+        internal static ReadOnlySpan<T> AsSpan<T>(this ImmutableArray<T> array)
```

Comment:
Does it make sense to have these be `in`?

Reply:
➡️ Answered in https://github.com/dotnet/roslyn/pull/31785#discussion_r241651338